packages/client/src/object/convertWireToOsdkObjects/createOsdkObject.ts

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright 2024 Palantir Technologies, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import type { OntologyObjectV2 } from "@osdk/internal.foundry";
+import type { MinimalClient } from "../../MinimalClientContext.js";
+import type { FetchedObjectTypeDefinition } from "../../ontology/OntologyProvider.js";
+import { Attachment } from "../Attachment.js";
+import { createClientCache } from "../Cache.js";
+import { get$as } from "./getDollarAs.js";
+import { get$link } from "./getDollarLink.js";
+import {
+  ClientRef,
+  ObjectDefRef,
+  RawObject,
+  UnderlyingOsdkObject,
+} from "./InternalSymbols.js";
+import type {
+  ObjectHolder,
+  ObjectHolderPrototypeOwnProps,
+} from "./ObjectHolder.js";
+import { OsdkCustomInspectPrototype } from "./OsdkCustomInspectPrototype.js";
+import type { PropertyDescriptorRecord } from "./PropertyDescriptorRecord.js";
+
+const objectPrototypeCache = createClientCache(
+  function(client, objectDef: FetchedObjectTypeDefinition<any, any>) {
+    return Object.create(
+      process.env.target !== "browser" ? OsdkCustomInspectPrototype : null,
+      {
+        [ObjectDefRef]: { value: objectDef },
+        [ClientRef]: { value: client },
+        "$as": { value: get$as(objectDef) },
+        "$link": {
+          get: function(this: ObjectHolder<typeof objectDef>) {
+            return get$link(this);
+          },
+        },
+        "$updateInternalValues": {
+          value: function(
+            this: ObjectHolder<typeof objectDef>,
+            newValues: Record<string, any>,
+          ) {
+            this[RawObject] = Object.assign(
+              {},
+              this[RawObject],
+              newValues,
+            );
+          },
+        },
+      } satisfies PropertyDescriptorRecord<ObjectHolderPrototypeOwnProps>,
+    );
+  },
+);
+
+export function createOsdkObject<
+  Q extends FetchedObjectTypeDefinition<any, any>,
+>(
+  client: MinimalClient,
+  objectDef: Q,
+  rawObj: OntologyObjectV2,
+) {
+  const kk = Object.create(objectPrototypeCache.get(client, objectDef), {
```

Comment:
super nit: could we use a clearer variable name please?

Reply:
Appreciate you keeping me honest :)